src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -1849,6 +1890,49 @@ private void RemoveStream(Http2Stream http2Stream)
             _concurrentStreams.AdjustCredit(1);
         }
 
+        private void RefreshPingTimestamp()
+        {
+            _nextPingRequestTimestamp = Environment.TickCount64 + (long)_keepAlivePingDelay.TotalMilliseconds;
+        }
+
+        private void ProcessPingAck(long payload)
+        {
+            if (_keepAliveState != KeepAliveState.PingSent)
+                ThrowProtocolError();
+            if (Interlocked.Read(ref _keepAlivePingPayload) != payload)
+                ThrowProtocolError();
+            _keepAliveState = KeepAliveState.None;
+            RefreshPingTimestamp();
```

Comment:
Is `RefreshPingTimestamp` required here? The timestamp should have already been refreshed in `ProcessIncomingFramesAsync`

Reply:
you are right